test/powershell/Modules/Microsoft.PowerShell.Core/Pester.Commands.Cmdlets.GetCommand.Tests.ps1

```diff
@@ -49,4 +49,52 @@ Describe "Tests Get-Command with relative paths and wildcards" -Tag "CI" {
         Pop-Location
     }
 
+    It "Get-Command -ShowCommandInfo property field test" {
+        $commandInfo = Get-Command Get-Date -ShowCommandInfo
+        $properties = ($commandInfo | Get-Member -MemberType NoteProperty)
+        $properties.Name.Contains("CommandType") | Should Be $true
+        $properties.Name.Contains("Definition") | Should Be $true
+        $properties.Name.Contains("Module") | Should Be $true
+        $properties.Name.Contains("ModuleName") | Should Be $true
+        $properties.Name.Contains("Name") | Should Be $true
+        $properties.Name.Contains("ParameterSets") | Should Be $true
+    }
+
+    $testcases = @(
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).Name}; testname = "Name"; result = "Get-Date"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).ModuleName}; testname = "ModuleName"; result = "Microsoft.PowerShell.Utility"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).Module.Name}; testname = "Module"; result = "Microsoft.PowerShell.Utility"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).CommandType}; testname = "CommandType"; result = "Cmdlet"}
+                  @{command = {(Get-Command Get-Date -ShowCommandInfo).Definition.Count}; testname = "Definition"; result = 1}
+               )
+
+    It "Get-Command -ShowCommandInfo property test - <testname>" -TestCases $testcases{
+            param (
+            $command,
+            $result
+        )
+        $command.Invoke() | Should Be $result
+    }
+
+
+    It "Get-Command -ShowCommandInfo ParameterSets property field test" {
+        $commandInfo = Get-Command Get-Date -ShowCommandInfo
+        $properties = ($commandInfo.ParameterSets[0] | Get-Member -MemberType NoteProperty)
+        $properties.Name.Contains("IsDefault") | Should Be $true
+        $properties.Name.Contains("Name") | Should Be $true
+        $properties.Name.Contains("Parameters") | Should Be $true
+    }
+
+    It "Get-Command -ShowCommandInfo Parameters property field test" {
+        $commandInfo = Get-Command Get-Date -ShowCommandInfo
```

Comment:
This should also be structured like the comment above.

Reply:
resolved